lua/entities/gmod_wire_egp/lib/egplib/transmitreceive.lua

```diff
@@ -625,13 +625,18 @@ if (SERVER) then
 						isLastScreen = nil
 					end
 
+					local data = {
+						Ent = v,
+						Objects = DataToSend,
+						Filtering = v.GPU_texture_filtering,
+						IsLastScreen = isLastScreen -- Doubles as notifying the client that no more data will arrive, and tells them how many did arrive
+					}
+
+					local von = WireLib.von.serialize(data)
+					local compressed = util.Compress(von)
 					net.Start("EGP_Request_Transmit")
-						net.WriteTable({
-							Ent = v,
-							Objects = DataToSend,
-							Filtering = v.GPU_texture_filtering,
-							IsLastScreen = isLastScreen -- Doubles as notifying the client that no more data will arrive, and tells them how many did arrive
-						})
+					net.WriteUInt( #compressed, 16 )
```

Comment:
I think we need an error or something to enforce a limit on the length of the WriteData